gremlin-integrations/templates/_helpers.tpl

```diff
@@ -0,0 +1,129 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "gremlin-integrations.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "gremlin-integrations.fullname" -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default .Chart.Name .Values.nameOverride }}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "gremlin-integrations.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Common labels
+*/}}
+{{- define "gremlin-integrations.labels" -}}
+helm.sh/chart: {{ include "gremlin-integrations.chart" . }}
+{{ include "gremlin-integrations.selectorLabels" . }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end }}
+
+{{/*
+Selector labels
+*/}}
+{{- define "gremlin-integrations.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "gremlin-integrations.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "gremlin-integrations.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create }}
+{{- default (include "gremlin-integrations.fullname" .) .Values.serviceAccount.name }}
+{{- else }}
+{{- default "default" .Values.serviceAccount.name }}
+{{- end }}
+{{- end }}
+
+{{- /* GKE Container Optimized OS with Containerd cannot */ -}}
+{{- /* mount the state/logs volumes, so detect that here */ -}}
+{{- define "gkeCOSContainerd" -}}
+{{- $output := false }}
+{{- range $index, $node := (lookup "v1" "Node" "" "").items -}}
+  {{- $gkeRuntime := index $node.metadata.labels "cloud.google.com/gke-container-runtime" -}}
+  {{- $gkeOS := index $node.metadata.labels "cloud.google.com/gke-os-distribution" -}}
+  {{- $output = (or $output (and (eq $gkeRuntime "containerd") (eq $gkeOS "cos"))) -}}
+{{- end -}}
+{{ $output }}
+{{- end -}}
```

Comment:
This is unused (as it should be, given integrations does not need to mount state/logs with a hostPath)
```suggestion
```